crates/matrix-sdk/src/sliding_sync/room.rs

```diff
@@ -167,7 +167,10 @@ impl SlidingSyncRoom {
                 inner.name = name;
             }
 
-            if avatar.is_some() {
```

Comment:
Surely this should be applied to `name`, `is_dm` etc, no?

Reply:
Good question, but I checked, and in my understanding, they don't have to:

- the `name` is always set
- `is_dm` uses a `Option<bool>` to determine both states, and would switch from `Some(true)` to `Some(false)` (or conversely) upon changes (`None` already meaning "hasn't changed since previous pos").

Reply:
<img width="493" alt="Screenshot 2023-11-21 at 14 21 09" src="https://github.com/matrix-org/matrix-rust-sdk/assets/637564/f7cd909e-1682-4034-ada6-6d62b755e90b">

I managed to completely remove my room's name so I think that's totally possible. 

Ack on the dm flag 👍

Reply:
> I managed to completely remove my room's name

In this case the proxy should generate a name based on the "heroes" of the room.

Reply:
Yep, I've confirmed this by tweaking the integration test to try that scenario.